src/disk_utils.cpp

```diff
@@ -1061,7 +1062,8 @@ int build_disk_index(const char *dataFilePath, const char *indexFilePath, const
                          "reorder (set true to include full precision in data file"
                          ": optional paramter, use only when using disk PQ\n"
                          "build_PQ_byte (number of PQ bytes for inde build; set 0 to use "
-                         "full precision vectors)"
+                         "full precision vectors)\n"
+                         "QD Quantized Dimension to overwrite the derived dim from B "
```

Comment:
> should we make this one or the other? Either you can specify memory budget or you can just #quantized bytes but not both?

what I intend to do here is to use -QD (quantized dim) to overwrite the existing derived pq dimension from-B, this is more backswords compactable. and do not need to force the prod DLVS diskann index build to pass in just one of the params

Reply:
The existing pipelines will be fine since we still allow B-only parameter.  For new pipelines, allowing both QD and B is confusing. If QD is greater than what B allows, will quantization be QD or B. I prefer that we allow user to specify B or QD and not both together

Reply:
> The existing pipelines will be fine since we still allow B-only parameter. For new pipelines, allowing both QD and B is confusing. If QD is greater than what B allows, will quantization be QD or B. I prefer that we allow user to specify B or QD and not both together

added check to disable both params pass in